Marlin/pins_5DPRINT.h

```diff
@@ -63,19 +63,12 @@
 #define TEMP_0_PIN          1  // Extruder / Analog pin numbering
 #define TEMP_BED_PIN        0  // Bed / Analog pin numbering
 
-// The SDSS pin uses a different pin mapping from file Sd2PinMap.h
+// The SDSS pin uses a different pin mapping from file fastio.h
 #define SDSS               20
 
-#if DISABLED(SDSUPPORT)
-  // these pins are defined in the SD library if building with SD support
-  #define SCK_PIN           9
-  #define MISO_PIN         11
-  #define MOSI_PIN         10
-#endif
-
 // Microstepping pins
 // Note that the pin mapping is not from fastio.h
-// See Sd2PinMap.h for the pin configurations
+// See Sd2PinMap.h for the pin configurations ???
```

Comment:
There is no concept like "straight digital pin numbers" https://github.com/MarlinFirmware/Marlin/issues/2970#issuecomment-217200401.
With this PR we only eliminate one of three different methods to map the pins.

Pimped up DirectIO a bit (https://github.com/AnHardt/Marlin/tree/DirectIO). This, indeed, has potential to unify all pin mappings. If we can find macros to replace the fastio.h-macros READ(), WRITE(), and so on, the change in Marlin would only be to replace the remaining calls to the Arduino functions (digitalRead(), digitalWrite(), analogeWrite(), ...) analogRead() shuld not occure. And then let's hope DirectIO is not much faster/slower than fastio.
The original DirectIO weas fulfilling almost all of my requirements - except the support for all boards. Hopefully i could change that.